map.go

```diff
@@ -28,42 +28,6 @@ var (
 	errMapNoBTFValue    = errors.New("map spec does not contain a BTF Value")
 )
 
-type IncompatibleMapErr struct {
-	Existing *MapInfo
```

Comment:
@ti-mo why did you get rid of *MapInfo and *MapSpec? To me they seem like the logical types to use for this.

Maybe `Existing` should even be `*Map` so that user code can move KVs around?

Reply:
Ergonomics, mainly. `*Map` has all of its properties hidden behind accessors, so at the call site this will end up as `if merr.Existing.Type() == merr.Incoming.Type { ... }`. I have a slight preference for a symmetrical comparison.

Your suggestion of `Existing *Map` makes more sense to me than `Existing *MapInfo`. A small downside is that we'll be sending open fds to the caller that will be left for the GC to clean up, guaranteed. Would be pretty convenient for writing migration code, though.

Reply:
> I have a slight preference for a symmetrical comparison.

Me too, but at this point we don't always have symmetrical info. MapSpec tends to have more info than what we can extract from a kernel side map due to missing APIs. I think the best course is to not obfuscate this.

> A small downside is that we'll be sending open fds to the caller that will be left for the GC to clean up, guaranteed.

Hmm good point about fds. We've worked so hard to get rid of the leaks xD. Maybe expose a function that allows loading the pinned map? Or a way to achieve something similar?

Reply:
I'm warming up to the idea of returning `struct IncompatibleMapError { Incoming *MapSpec, Existing *Map}` and `errors.Join()`ing them during ELF load. It's not expected to trigger this error very often, so the impact of leaking *Maps is relatively limited. This basically implements your `MigrateMap func(...) error` idea, since it exposes a mutable reference to the offending MapSpec and allows manipulating the existing Map as well.

This would also play nicely with a potential feature proposal I have for overriding bpffs pin paths for specific maps:

```
CollectionOptions{
	Maps: MapOptions{PinPath: "/sys/fs/bpf"},
	MapPinPaths: map[string]string{
		"cilium_calls": "/sys/fs/bpf/some/nested/dir/cilium_calls",
	},
}
```

This way, the caller just receives a *Map from an `IncompatibleMapError` and doesn't need to care about where the pin was on the file system.